include/mbgl/storage/default_file_source.hpp

```diff
@@ -2,9 +2,135 @@
 #define MBGL_STORAGE_DEFAULT_FILE_SOURCE
 
 #include <mbgl/storage/file_source.hpp>
+#include <mbgl/util/geo.hpp>
+
+#include <vector>
 
 namespace mbgl {
 
+/*
+ * An offline region defined by a style URL, geographic bounding box, zoom range, and
+ * device pixel ratio.
+ */
+class OfflineTilePyramidRegionDefinition {
+public:
+    OfflineTilePyramidRegionDefinition(const std::string&, const LatLngBounds&, double, double, float);
```

Comment:
The intent is that they can be used to specify a subrange of the full zoom range. If you want the full range supported by sources, pass `minZoom = 0` and `maxZoom = Infinity`. I'm fine if the SDK bindings use those as default values.